### PR TITLE
Add documentation for changes introduced by jslib-aws v0.7.0

### PR DESCRIPTION
**Pull request by oleiade**
This PR documents the changes introduced by [jslib-aws v0.7.0](https://github.com/grafana/k6-jslib-aws/releases/tag/v0.7.0).

Let me know what you think 🙇🏻

### CODE REVIEW

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md
Comment (MattDodsonEnglish) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md:31]:
```suggestion
SignatureV4 methods throw errors on failure.
```

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md
Comment (MattDodsonEnglish) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md:8]:
```suggestion
With SignatureV4, you can  produce authenticated HTTP requests to AWS services. Namely, it lets you to sign and pre-sign requests to AWS services using the [Signature V4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) algorithm. The `sign` operation produces a signed request, with authorization information stored in its headers, while the `presign` operation produces a pre-signed request, with authorization information stored in its query string parameters.
```

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md
Comment (MattDodsonEnglish) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md:10]:
```suggestion
SignatureV4 is included in both the dedicated jslib `signature.js` bundle and the `aws.js` one, containing all the service's clients.
```

**Review by MattDodsonEnglish — COMMENTED** [review]
Swooped through with some nits. There also needs to be some standardization of "presign" or "pre-sign".

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/02 presign().md
Comment (oleiade) [reply src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/02 presign().md:28]: Hehe I did a pass before pushing the PR, but it seems some went through still, will try to catch the rest indeed 🤞🏻 🙇🏻

**Comment by oleiade** [comment]
Hey @MattDodsonEnglish did you get a chance to lay your eyes on this again? You always find some nifty gritty stuff 😍 , so I'd rather wait for your thumbs up before merging 🙇🏻

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md
Comment (MattDodsonEnglish) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md:12]:
```suggestion
With SignatureV4, you can produce authenticated HTTP requests to AWS services. Namely, it lets you sign and pre-sign requests to AWS services using the [Signature V4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) algorithm. The `sign` operation produces a signed request with authorization information stored in its headers.
The `presign` operation produces a pre-signed request with authorization information stored in its query string parameters.

SignatureV4 is included in both the dedicated jslib `signature.js` bundle and the `aws.js` one, containing all the service's clients.

Instantiating a new `SignatureV4` requires a single options object argument with the following properties: 
```

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md
Comment (MattDodsonEnglish) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md:19]:
```suggestion
| uriEscapePath | boolean                                                                                   | Whether to uri-escape the request URI path as part of computing the canonical request string. As of late 2017, this is **required for every AWS service** except Amazon S3.                                                                                                |
```

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md
Comment (MattDodsonEnglish) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md:20]:
```suggestion
| applyChecksum | boolean                                                                                   | Whether to calculate a checksum of the request body and include it as either a request header (when signing) or as a query string parameter (when pre-signing). This is **required for AWS Glacier and Amazon S3** and optional for every other AWS service as of late 2017. |
```

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/01 sign().md
Comment (MattDodsonEnglish) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/01 sign().md:24]:
```suggestion
You can override SignatureV4 options in the context of this specific request. To do this, pass a second parameter to the `sign` method, which is an Object with the following parameters.
```

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/02 presign().md
Comment (MattDodsonEnglish) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/02 presign().md:7]:
```suggestion
`SignatureV4.presign()` pre-signs a URL with the AWS Signature V4 algorithm. Given an HTTP request description, it returns a new HTTP request with the AWS signature v4 authorization added. It returns an Object holding a `url` containing the authorization information encoded in its query string, ready to use in the context of a k6 HTTP call.
```

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/02 presign().md
Comment (MattDodsonEnglish) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/02 presign().md:24]:
```suggestion
You can provide further options and override SignatureV4 options in the context of this specific request.
To do this, pass a second parameter to the `presign` method, which is an Object with the following parameters.
```

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/02 presign().md
Comment (MattDodsonEnglish) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/02 presign().md:41]:
```suggestion
| headers  | Object | The pre-signed request headers to use in the context of a k6 HTTP request |
```

**Review by MattDodsonEnglish — APPROVED** [review]
Looks good! Approving with some nits, including:
- Reducing passive voice
- Bolding required fields
- Removing possessive, I think it comes off a little anthropomorphic here. Turning the service into a non-possessive adjective works better, IMO. And it saves two characters:
   ```diff
   - The client's secret
   + The client secret
   ```
- Fixing some things that I broke :grimacing:

**Comment by oleiade** [comment]
Thanks a lot for your further review and changes @MattDodsonEnglish much appreciated! 🙇🏻 I do have some tendency to overuse possessives for sure!

**Pull request merged**
